packages/deployments/contracts/contracts/core/messaging/connectors/Connector.sol

```diff
@@ -222,21 +236,51 @@ abstract contract Connector is ProposedOwnable, MerkleTreeManager, ConnectorMana
   }
 
   // ============ Public fns ============
-  function sendMessage(bytes memory _data) external {
-    _sendMessage(_data);
+  /**
+   * @notice This should be called by RootManager to propagate an updated aggregate root on the Hub/L1
+   * chain out to the spoke/L2 chains.
+   */
+  function propagateRoot(bytes memory _data) external onlyRootManager onlyHub {
+    // RootManager has aggregated roots from all other Connectors on this hub chain.
+    // TODO: Should we update our current aggregateRoot to reflect the new one.
+    // aggregateRoot = bytes32(_data);
+
+    // TODO: This code block is partially redundant with the `sendRoot` method. Should we just make `sendRoot` public?
```

Comment:
`sendRoot` is public? im confuse

Reply:
it's `external` and `external != public` lol.

I was thinking - should we just make sendRoot public, then we can technically call it externally *and* internally but meh